Rules.py

```diff
@@ -66,5 +66,62 @@ def global_rules(world):
+    # location names used are mostly guesses and can absolutely be changed later
+    # item names used are also guesses, as are some state function names probly
+    # location names may be longer than necessary so as to be descriptive, like we can change 'South Clock Town Hookshot
+    # Ledge Rupee Chest' if we want lol
+    set_rule(world.get_location('Clock Tower Platform HP'), lambda state: state.form('Human') or state.form('Goron') or state.form('Zora') or state.can_reach(world.get_location('Clock Town Business Scrub')))
+    set_rule(world.get_location('Festival Tower Rupee Chest'), lambda state: state.has('Hookshot') or state.can_reach(world.get_location('Clock Town Business Scrub')))
+    set_rule(world.get_location('South Clock Town Hookshot Ledge Rupee Chest'), lambda state: state.has('Hookshot'))
+    set_rule(world.get_location('Bremen Mask From Guru Guru'), lambda state: state.form('Human'))
+    set_rule(world.get_location('Rosa Sisters HP'), lambda state: state.has('Kamaro Mask'))
+
+    # obviously, this and simliar 'tests' might not be necessary at all
+    set_rule(world.get_location('Adult Wallet from bank'), lambda state: True)
+    set_rule(world.get_location('Bank HP'), lambda state: True)
+
+    # I don't know that this item matters at all or if we should even have a spot for it
+    # but we can if we want, you get it the same way you get the code in the first cycle, you just have to be human to
+    # get the actual notebook from them I think
+    set_rule(world.get_location('Bomber Notebook'), lambda state: state.has('Magic Meter') and state.form('Deku') and state.form('Human'))
+
+    # since getting the blast mask just involves slashing sakon, it might be possible to get it with other forms? like
+    # maybe zora? probly not goron though
+    # also you might need to be link to talk to the lady after, so maybe it is only human
+    set_rule(world.get_location('Blast Mask'), lambda state: state.form('Human'))
+
+    set_rule(world.get_location('North Clock Town Tree HP'), lambda state: True)
+    set_rule(world.get_location('Tingle Clock Town Map'), lambda state: True)
+    set_rule(world.get_location('Tingle Woodfall Map'), lambda state: True)
+    set_rule(world.get_location('Keaton HP'), lambda state: state.has('Keaton Mask'))
+
+    # I need to figure out all the actual requirements for this, including tricks, so this one is tentative
+    # also the trick names are guesses for sure
+    set_rule(world.get_location('East Clock Town 100 Rupee Chest'), lambda state: state.form('Human') or state.can('Goron Boost') or state.can('Gainer'))
+
+    set_rule(world.get_location('Ocarina of Time'), lambda state: (state.form('Deku') and state.has('Magic Meter')) or (state.has('Bow') and state.form('Human')))
+    set_rule(world.get_location('All Night mask'), lambda state: state.form('Human') and state.has('Giants Wallet'))
+    set_rule(world.get_location('Bigger Bomb Bag'), lambda state: state.has('Giants Wallet'))
```

Comment:
And `has('Bomb Mask')`, since you have to AT LEAST once get it to get the Bigger Bomb Bag into the shop.